Applications/Spire/Include/Spire/Spire/Utility.hpp

```diff
@@ -1,8 +1,21 @@
 #ifndef SPIRE_UTILITY_HPP
 #define SPIRE_UTILITY_HPP
+#include <QObject>
 
 namespace Spire {
 
+  //! Deletes an object when control returns to the thread's event loop.
+  /*
+    \param object The object to schedule for deletion.
+  */
+  template<typename T>
+  std::enable_if_t<std::is_base_of_v<QObject, T>> deleteLater(T*& object) {
```

Comment:
It seems like snake_case should be used for free functions, and there won't be conflicts with `deleteLater` method so you can remove the `Spire::` prefix.